compact_str/src/tests.rs

```diff
@@ -370,6 +370,141 @@ fn test_plus_operator() {
     assert_eq!(String::from("a") + CompactStr::from("b"), "ab");
 }
 
```

Comment:
So we basically unit test the integer and float specialisation of `ToCompactStr` here.

Note that we also have specialisation for `NonZero*` newtype wrappers of the integer, `char`, `String` and `CompactStr`.

Reply:
Good call out for the `NonZero*` types, I added those to the fuzzing